### PR TITLE
Added option for dumping resource views into the resource dicts durin…

### PR DESCRIPTION
**Pull request by JVickery-TBS**
…g dataset dumps.

Added --resource-views (-R) flag to the dump datasets api action. The boolean flag allows for the resource views lists to be added into the resource dicts as the index "resource_views".

If there are no views for a resource, it will NOT add the "resource_views" index to the resource dicts.

Also added the new flag to the pytests.

### CODE REVIEW

File: ckanapi/datapackage.py [hunk 1]
```diff
@@ -137,6 +137,25 @@ def populate_datastore_res_fields(ckan, res):
         return  # with localckan we'll get the real CKAN exception not a CKANAPIError subclass
     res['datastore_fields'] = ds['fields']
 
+def populate_datastore_res_views(ckan, res):
```
Comment (wardi) [inline ckanapi/datapackage.py:140]: the `datapackage.py` module is about a specific thing called "datapackages" this function would be better in the `dump.py` module.

Comment (JVickery-TBS) [reply ckanapi/datapackage.py]: @wardi populate_datastore_res_views method has been moved to the dump.py module now.

**Comment by wardi** [comment]
Thanks @JVickery-TBS were you going to add a small test for the feature as well?

**Comment by wardi** [comment]
that's what I was thinking, something to show that the resource view data will actually be included in the output (and what format it will follow)

**Comment by wardi** [comment]
The tests on this repo are now running again if you want to merge from master

**Comment by JVickery-TBS** [comment]
> 



> that's what I was thinking, something to show that the resource view data will actually be included in the output (and what format it will follow)

@wardi the new pytest is in there now.

**Comment by wardi** [comment]
Nice. Thanks @JVickery-TBS

**Pull request merged**